torchaudio/datasets/yesno.py

```diff
@@ -17,10 +19,108 @@
         "folder_in_archive": "waves_yesno",
         "url": "http://www.openslr.org/resources/1/waves_yesno.tar.gz",
         "checksum": "30301975fd8c5cac4040c261c0852f57cfa8adbbad2ce78e77e4986957445f27",
+        "files": [
+            "0_0_0_0_1_1_1_1.wav",
+            "0_0_0_1_0_0_0_1.wav",
+            "0_0_0_1_0_1_1_0.wav",
+            "0_0_1_0_0_0_1_0.wav",
+            "0_0_1_0_0_1_1_0.wav",
+            "0_0_1_0_0_1_1_1.wav",
+            "0_0_1_0_1_0_0_0.wav",
+            "0_0_1_0_1_0_0_1.wav",
+            "0_0_1_0_1_0_1_1.wav",
+            "0_0_1_1_0_0_0_1.wav",
+            "0_0_1_1_0_1_0_0.wav",
+            "0_0_1_1_0_1_1_0.wav",
+            "0_0_1_1_0_1_1_1.wav",
+            "0_0_1_1_1_0_0_0.wav",
+            "0_0_1_1_1_0_0_1.wav",
+            "0_0_1_1_1_1_0_0.wav",
+            "0_0_1_1_1_1_1_0.wav",
+            "0_1_0_0_0_1_0_0.wav",
+            "0_1_0_0_0_1_1_0.wav",
+            "0_1_0_0_1_0_1_0.wav",
+            "0_1_0_0_1_0_1_1.wav",
+            "0_1_0_1_0_0_0_0.wav",
+            "0_1_0_1_1_0_1_0.wav",
+            "0_1_0_1_1_1_0_0.wav",
+            "0_1_1_0_0_1_1_0.wav",
+            "0_1_1_0_0_1_1_1.wav",
+            "0_1_1_1_0_0_0_0.wav",
+            "0_1_1_1_0_0_1_0.wav",
+            "0_1_1_1_0_1_0_1.wav",
+            "0_1_1_1_1_0_1_0.wav",
+            "0_1_1_1_1_1_1_1.wav",
+            "1_0_0_0_0_0_0_0.wav",
+            "1_0_0_0_0_0_0_1.wav",
+            "1_0_0_0_0_0_1_1.wav",
+            "1_0_0_0_1_0_0_1.wav",
+            "1_0_0_1_0_1_1_1.wav",
+            "1_0_1_0_1_0_0_1.wav",
+            "1_0_1_1_0_1_1_1.wav",
+            "1_0_1_1_1_0_1_0.wav",
+            "1_0_1_1_1_1_0_1.wav",
+            "1_1_0_0_0_0_0_1.wav",
+            "1_1_0_0_0_1_1_1.wav",
+            "1_1_0_0_1_0_1_0.wav",
+            "1_1_0_0_1_0_1_1.wav",
+            "1_1_0_0_1_1_1_0.wav",
+            "1_1_0_1_0_1_0_0.wav",
+            "1_1_0_1_0_1_1_0.wav",
+            "1_1_0_1_1_0_0_1.wav",
+            "1_1_0_1_1_0_1_1.wav",
+            "1_1_0_1_1_1_1_0.wav",
+            "1_1_1_0_0_0_0_1.wav",
+            "1_1_1_0_0_1_0_1.wav",
+            "1_1_1_0_0_1_1_1.wav",
+            "1_1_1_0_1_0_1_0.wav",
+            "1_1_1_0_1_0_1_1.wav",
+            "1_1_1_1_0_0_1_0.wav",
+            "1_1_1_1_0_1_0_0.wav",
+            "1_1_1_1_1_0_0_0.wav",
+            "1_1_1_1_1_1_0_0.wav",
+            "1_1_1_1_1_1_1_1.wav",
+        ]
     }
 }
 
 
+@dataclass
+class YesNoItem:
+    path: str
+    label: List[int]
+    waveform: Tensor
+    sample_rate: int
+
+
+class ListYesNoItems(IterDataPipe):
+    """Given a root directory, return the list of files"""
+    def __init__(self, root):
+        self.data_dir = os.path.join(root, 'waves_yesno')
+        self.files = _RELEASE_CONFIGS['release1']['files']
+
+    def __iter__(self):
+        for filename in self.files:
+            path = os.path.join(self.data_dir, filename)
+            label = [int(c) for c in path.split("_")]
+            yield path, label
```

Comment:
Can you use [`ListDirFiles`](https://github.com/pytorch/pytorch/blob/f364e492dfb287e4043d37fffa1bcef55e2ac5dd/torch/utils/data/datapipes/iter/listdirfiles.py#L5) to generate a filename per iteration.